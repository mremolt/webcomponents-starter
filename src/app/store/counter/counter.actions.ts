import { createAction } from 'typesafe-actions';

export const increment = createAction('[Counter] Increment');

export const decrement = createAction('[Counter] Decrement');

export const add = createAction('[Counter] Add', action => {
  return (amount: number) => action(amount);
});
