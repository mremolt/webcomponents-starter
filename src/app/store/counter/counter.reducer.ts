import { ActionType, getType } from 'typesafe-actions';
import { DeepReadonly } from 'utility-types';

import * as actions from './counter.actions';

export type CounterAction = ActionType<typeof actions>;

export const initialState = { value: 0 };

export type CounterState = DeepReadonly<typeof initialState>;

export const counterReducer = (state: CounterState = initialState, action: CounterAction): CounterState => {
  switch (action.type) {
    case getType(actions.increment):
      return { ...state, value: state.value + 1 };

    case getType(actions.decrement):
      return { ...state, value: state.value - 1 };

    case getType(actions.add):
      return { ...state, value: state.value + action.payload };

    default:
      return state;
  }
};
