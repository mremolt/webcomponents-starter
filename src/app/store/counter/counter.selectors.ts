import { createSelector } from 'reselect';

import { RootState } from '../root.reducer';

export const selectCounterState = (state: RootState): RootState['counter'] => state.counter;

export const selectValue = createSelector(
  [selectCounterState],
  state => state.value
);
