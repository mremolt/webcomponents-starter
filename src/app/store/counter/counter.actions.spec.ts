import { add, decrement, increment } from './counter.actions';

describe('counter actions', () => {
  describe('increment', () => {
    it('returns the correct action', () => {
      expect(increment()).toMatchSnapshot();
    });
  });

  describe('decrement', () => {
    it('returns the correct action', () => {
      expect(decrement()).toMatchSnapshot();
    });
  });

  describe('add', () => {
    it('returns the correct action', () => {
      expect(add(2)).toMatchSnapshot();
    });
  });
});
