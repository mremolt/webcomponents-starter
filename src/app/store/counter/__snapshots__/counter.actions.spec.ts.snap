// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter actions add returns the correct action 1`] = `
Object {
  "error": undefined,
  "meta": undefined,
  "payload": 2,
  "type": "[Counter] Add",
}
`;

exports[`counter actions decrement returns the correct action 1`] = `
Object {
  "error": undefined,
  "meta": undefined,
  "payload": undefined,
  "type": "[Counter] Decrement",
}
`;

exports[`counter actions increment returns the correct action 1`] = `
Object {
  "error": undefined,
  "meta": undefined,
  "payload": undefined,
  "type": "[Counter] Increment",
}
`;
